static/app/views/issueList/draggableTabBar.tsx

```diff
@@ -1,15 +1,28 @@
 import 'intersection-observer'; // polyfill
 
 import {useState} from 'react';
-import type {Key} from '@react-types/shared';
+import styled from '@emotion/styled';
+import type {Key, Node} from '@react-types/shared';
 
+import Badge from 'sentry/components/badge/badge';
 import {DraggableTabList} from 'sentry/components/draggableTabs/draggableTabList';
+import type {DraggableTabListItemProps} from 'sentry/components/draggableTabs/item';
+import type {MenuItemProps} from 'sentry/components/dropdownMenu';
+import QueryCount from 'sentry/components/queryCount';
 import {TabPanels, Tabs} from 'sentry/components/tabs';
+import {space} from 'sentry/styles/space';
+import {DraggableTabMenuButton} from 'sentry/views/issueList/draggableTabMenuButton';
 
 export interface Tab {
   content: React.ReactNode;
   key: Key;
   label: string;
+  hasUnsavedChanges?: boolean;
+  onDelete?: (key: MenuItemProps['key']) => void;
```

Comment:
Since these action callbacks accept a `key` parameter, they don't need to live on the `Tab` object, right? It will probably be simpler and less error-prone to  pass these in `DraggableTabProps` instead.

Reply:
You're completely right, good catch!